go/proof.go

```diff
@@ -313,13 +318,17 @@ func getPadding(spec *InnerSpec, branch int32) (minPrefix, maxPrefix, suffix int
 // on the left side of this branch, ie. it's a valid placeholder on a leftmost path
 func leftBranchesAreEmpty(spec *InnerSpec, op *InnerOp, branch int32) bool {
 	idx := getPosition(spec.ChildOrder, branch)
-	// compare the prefix bytes with the appropriate number of empty children
-	leftChildren := idx
-	actualPrefix := len(op.Prefix) - leftChildren*int(spec.ChildSize)
+	// count branches to left of this
+	leftBranches := idx
+	if leftBranches == 0 {
+		return false
```

Comment:
That's an odd choice.
If there are no left branches, then all left branches are empty, right?
Why return false here?

(The true case would fall out after the loop anyway, but fine to explicitly return early. Not sure why you want to return false here: `on the left side of this branch, ie. it's a valid placeholder on a leftmost path` implies that no leftBranches at all would satisfy this)

Reply:
When the index is 0, the op describes a left-branch, but the op contains the data of the siblings of that branch, right? So `leftBranches` is the count of branches to the left of the "described" branch, which is none. It's not really a valid input to the function, because it asks a meaningless question: "are the branches to the left of this one empty?" That's how I see it, so an error may be more appropriate, but returning false seemed better than true.

Reply:
again, as this function is only used to check "leftmost" of a path, I would do this differently. But no need to repeat myself